### PR TITLE
enhance Custom Client example

### LINKED ISSUE #176: enhance Custom Client example
Our current example code does not handle errors correctly:

```
export class SampleAPI {
  token: string
  constructor(params: { token: string }) {
    this.token = params.token
    this.baseUrl = 'http://localhost:3333/api'
  }
  client = () => {
    return initClient(userContract, {
      baseUrl: this.baseUrl,
      baseHeaders: {
        Authorization: `Bearer ${idToken}`,
        'Content-Type': 'application/json'
      },
      api: async ({ path, method, headers, body }) => {
        const result = await axios.request({
          method: method as Method,
          url: `${this.baseUrl}/${path}`,
          headers,
          data: body,
        }) // <--- we should catch the axios error and pass it upstream for the user to handle
        return { status: result.status, body: result.data }
      },
    })
  }
}
```
React Query somehow recognizes an error happens (probably because the promise does not resolve). Still, when the error's data isn't passed, we cannot access any data (or typesafe data). 

We should signal users to catch and return data appropriately.

### PR DESCRIPTION
**Pull request by michaelangrivera**
fixes #176

### CODE REVIEW

File: apps/docs/docs/core/custom.md [hunk 1]
```diff
@@ -15,20 +15,31 @@ you can pass a `api` attribute to the `initClient` or `initQueryClient`.
 
 **Here's a basic example: **
 ```typescript
+import { contract } from './some-contract'
+import axios, { Method, AxiosError, AxiosResponse, isAxiosError } from 'axios'
 const client = initClient(contract, {
```
Comment (Gabrola) [inline apps/docs/docs/core/custom.md:20]:
```suggestion
import axios, { Method, AxiosError, AxiosResponse, isAxiosError } from 'axios'

const client = initClient(contract, {
```

File: apps/docs/docs/core/custom.md [hunk 2]
```diff
@@ -15,20 +15,31 @@ you can pass a `api` attribute to the `initClient` or `initQueryClient`.
...
+      })
+      return { status: result.status, body: result.data }
+    } catch (e: Error | AxiosError | any) {
+      if (isAxiosError(e)) {
+        const error = e as AxiosError
+        const response = error.response as AxiosResponse
+        return { status: response.status, body: response.data }
+      }
+    }
+    return { status: 500, body: {} }
```
Comment (Gabrola) [inline apps/docs/docs/core/custom.md:42]: I think it makes sense to re-throw the error if it is not an axios error

```suggestion
      }
      
      throw e;
    }
```

File: apps/docs/docs/core/custom.md [hunk 3]
```diff
@@ -38,27 +49,38 @@ Sometimes you need dynamic headers, IE passing in a Bearer token. There are two
 ### Instantiate the `client` with the header passed in:
 
 ```typescript
+import { contract } from './some-contract'
+import axios, { Method, AxiosError, AxiosResponse, isAxiosError } from 'axios'
 export class SampleAPI {
```
Comment (Gabrola) [inline apps/docs/docs/core/custom.md:54]:
```suggestion
import axios, { Method, AxiosError, AxiosResponse, isAxiosError } from 'axios'

export class SampleAPI {
```

File: apps/docs/docs/core/custom.md [hunk 4]
```diff
@@ -70,33 +92,47 @@ export class SampleAPI {
...
       },
     })
   }
 }
 ```
 
 :::
+
+
+   Authorization: `Bearer ${token}`
```
Comment (Gabrola) [inline apps/docs/docs/core/custom.md:138]: Accidental copy-pasta?

**Review by Gabrola — COMMENTED** [review]
LGTM! Left a couple of nits

File: apps/docs/docs/core/custom.md (hunk 4 above)
Comment (michaelangrivera) [reply apps/docs/docs/core/custom.md:138]: lol whoops

File: apps/docs/docs/core/custom.md (hunk 2 above)
Comment (michaelangrivera) [reply apps/docs/docs/core/custom.md:42]: Yup, you're right; I was worried about the return type conflicting, but TS picks it up correctly.

**Review by oliverbutler — APPROVED** [review]

**Pull request merged**